internal/testutils/xds/e2e/clientresources.go

```diff
@@ -526,10 +526,22 @@ func ClusterResourceWithOptions(opts ClusterOptions) *v3clusterpb.Cluster {
 
 // LocalityOptions contains options to configure a Locality.
 type LocalityOptions struct {
-	// Ports is a set of ports on "localhost" belonging to this locality.
-	Ports []uint32
+	// Name is the unique locality name.
+	Name string
 	// Weight is the weight of the locality, used for load balancing.
 	Weight uint32
+	// Backends is a set of backends belonging to this locality.
+	Backends []BackendOptions
+}
+
+// BackendOptions contains options to configure individual backends in a
+// locality.
+type BackendOptions struct {
+	// Port number on which the backend is accepting connections.
+	Port uint32
+	// Health status of the backend. Default is UNKNOWN which is treated the
+	// same as HEALTHY.
+	HealthStatus v3corepb.HealthStatus
```

Comment:
Nit: Still looks like on Localhost. Mention somewhere in these comments? I guess maybe we'll scale it up in future, but then you can't test locally haha.

Reply:
Thanks. Added.

Yes, we probably will never make it generic enough to run on non-localhost machines :)